packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -207,27 +199,22 @@ export function createStylesheet(vm: VM, stylesheets: string[]): VNode | null {
     const { renderMode, shadowMode } = vm;
     if (renderMode === RenderMode.Shadow && shadowMode === ShadowMode.Synthetic) {
         for (let i = 0; i < stylesheets.length; i++) {
-            insertGlobalStylesheet(stylesheets[i]);
+            insertStylesheet(stylesheets[i]);
         }
-    } else if (ssr || isHydrating()) {
+    } else if (ssr || vm.hydrated) {
```

Comment:
One downside of this is that it will slightly bloat the bundle size for environments without hydration. In the previous version of the code, if the `hydrateComponent` API is unused, then this becomes `false` and the entire `else if` block is tree-shaken, including the `createInlineStyleVNode` function.

In practice, this is like ~5 lines of code, so I don't think it's a big deal. Also, a sufficiently smart minifier _could_ understand that `vm.hydrated` is always false, so we could potentially fix it in the future using a refactor.